lib/LaTeXML/Core/Mouth/file.pm

```diff
@@ -63,11 +65,17 @@ sub getNextLine {
       close($fh); $$self{IN} = undef;
       return; }
     else {
+      if (!$$self{encoding}) {
+        my $detected = detect($line);
+        if ($detected && $detected eq 'UTF-8') {
+          $$self{encoding} = 'UTF-8';
```

Comment:
I also ensured that this check will be performed only until the first UTF-8 line is spotted for a given file.

Sadly if a file is plain ascii (as all .sty files are for example), this `detect()` call will be invoked on every single line. So it better be cheap.